R/functions.R

```diff
@@ -3,6 +3,34 @@
 #' Return id of the planning googlesheet
 gs_id <- function() "1HLtyGK_csi5W_v7XChxgTuVjS-RKXqc0Jxos1RBqpwk"
 
+
+#' Sanity checks for the google sheet.
+#' This function is used in data pipe-lines to check that everything
+#' in the google sheet is as expected.
+#' It will throw errors or warnings if something goes wrong.
+#'
+#' @details the function can be used in a dplyr pipe, which is why
+#' it requires and returns the `.data` argument.
+#' @param .data a data frame (or derivative) loaded from a google sheet
+sanity_checks <- function (.data) {
+  # https://github.com/hadley/assertthat/issues/41
+
+  # check if there are NA columns not supposed to be empty
+  required_columns <- c('Thema', 'Subthema', 'Start')
+  if (any(is.na(.data[, required_columns]))) {
+    troublemakers = rownames(.data)[rowSums(is.na(.data[, required_columns])) > 0]
+    message(paste0("Start and Deadline of a task may not be empty! (line/s ", troublemakers, ")")
```

Comment:
Seems fair enough to use `warning()` instead of `message()`?

Reply:
`warning()` stopped the script; I googled and found this advice to use "message" instead:
https://reside-ic.github.io/blog/a-warning-about-warning/
... but my prior is `NA`, so I am happy for advice!

Reply:
Some other advice here: https://adv-r.hadley.nz/conditions.html?q=warning#warnings

I'd rather use a warning in this case, but have no strong opinion.

Reply:
I switched to `warning`, thank you for the link!